app/forms/mappings/Constraints.scala

```diff
@@ -81,6 +81,22 @@ trait Constraints {
         Invalid(errorKey, minimum)
     }
 
+  protected def numericAndCorrectLength(minimum: Int, maximum: Int): Constraint[String] =
+    Constraint {
+      case str if isNumeric(str) && hasCorrectLength(str, minimum, maximum)  =>
+        Valid
+      case str if isNumeric(str) && !hasCorrectLength(str, minimum, maximum) =>
+        Invalid("commodityCode.error.length")
+      case _                                                                 =>
+        Invalid("commodityCode.error.nonNumeric")
+    }
+
+  private def isNumeric(str: String): Boolean =
+    str.filterNot(_.isWhitespace).forall(_.isDigit)
```

Comment:
```suggestion
private def isNumeric(str: String): Boolean =
  str.replaceAll("\\s", "").forall(_.isDigit)
```

Reply:
Again, just a suggestion but I think it's easier to understand (personal choice)